File: mobile_app/lib/data/models/task.dart
import 'package:meta/meta.dart';

class Task {
  final int id;
  final String content;
  final int parentId;
  final int createdAt;
  final bool isCompleted;

  Task({
    @required this.id,
    @required this.content,
    @required this.parentId,
    @required this.createdAt,
    @required this.isCompleted,
  });
}

File: web_app/src/data/types/task.ts
export interface Task {
  id: string;
  content: string;
  createdAt: number;
  isComplete: boolean;
  parentId: string;
  hasSubtasks: boolean;
}

export type AddTaskData = Omit<
  Task,
  "id" | "createdAt" | "isComplete" | "hasSubtasks"
>;

export type UpdateTaskData = Partial<Omit<Task, "id">>;
